src/javascript/binary/pages/trade/message.js

```diff
@@ -3,33 +3,97 @@
  * events based on type of response
  */
 var Message = (function () {
-    'use strict';
+    "use strict";
 
     var process = function (msg) {
-        var response = JSON.parse(msg.data);
-        if (response) {
-            var type = response.msg_type;
-            if (type === 'authorize') {
-                TUser.set(response.authorize);
-                TradeSocket.send({ payout_currencies: 1 });
-            } else if (type === 'active_symbols') {
+
+        var response;
+
+        try {
+            response  = JSON.parse(msg.data);
+            if("object" !== typeof response || !("msg_type" in response)) {
+                throw new Error("Response from WS API is not well formatted.");
+            }
+        } catch(e) {
+            throw new Error("Response from WS API is not well formatted.");
```

Comment:
That is very weird code. Throw + recatch.
Also I don't think too strict format checking does any good, if you have no concrete plan to process the error.